src/core/server/http/router/request.ts

```diff
@@ -184,6 +190,8 @@ export class KibanaRequest<
 
     const options = ({
       authRequired: request.route.settings.auth !== false,
+      // some places in LP call KibanaRequest.from(request) manually. remove fallback to true before v8
+      xsrfRequired: (request.app as KibanaRouteState)?.xsrfRequired ?? true,
```

Comment:
__issue:__ I didn't manage to make this new feature work with real route and after some investigation it seems the issue is that we use the wrong `app`, I see that `xsrfRequest` is available in `request.route.settings.app` instead.

Aaand green CI makes me think that we need a real API integration test :)

Reply:
added test is incorrect, updated